### ISSUE TITLE
bug: Modify the route.yaml configuration to not update apisix synchronously

### ISSUE THREAD

**Issue by pioneer-hash** [apache/apisix-ingress-controller#312]
### Issue description
1.when I first  create apisix-route.yaml and excecute kubectl apply -f  apisix-route.yaml . Configure synchronization successful ,My configuration can be viewed in the ETCD.
2.  When I modify the apisix-route.yaml configuration for some parameters ,For example, change zhangsan to zhangsan111 , Unfortunately, the latest configuration has not been synchronized. 
this is my apisix-route.yaml 
```yaml
apiVersion: apisix.apache.org/v2alpha1
kind: ApisixRoute
metadata:
 name: test4
spec:
 http:
 - name: rule2
   match:
     paths:
       - /*
     exprs:
     - subject:
         scope: Cookie
         name: name
       op: Equal
       value: lisi
   backend:
     serviceName: nginx
     servicePort: 80
 - name: rule3
   match:
     paths:
       - /*
     exprs:
     - subject:
         scope: Cookie
         name: name
       op: Equal
       value: zhangsan
   backend:
     serviceName: ddb-svc
     servicePort: 8080
```

### Environment
 apisix-ingress-controller version is  the current master(2021/3/24 )

### What's the expected result?
 The apisix configuration is also updated when you modify some configuration in apisix-route.yaml and then execute kubectl apply-f apisix-route.yaml

**Comment by gxthrj** [comment]
Do you have any error logs in the container of INGRESS ?

**Comment by tokers** [comment]
It seems that the route synchronization logic is pathological, we may fix it.

**Comment by pioneer-hash** [comment]
> It seems that the route synchronization logic is pathological, we may fix it.

Looking forward to finishing soon.

**Comment by pioneer-hash** [comment]
> Do you have any error logs in the container of INGRESS ?

NO ,there are no error logs in the container of INGRESS

**Comment by tokers** [comment]
@pioneer-hash Thanks for your report, I have captured this bug, I will fix it as soon as possible.

**Comment by tokers** [comment]
Fixed by #319.
